### ISSUE TITLE
Support for SVG images

### ISSUE THREAD

**Issue by hrj** [UprootLabs/gngr#210]
Need to add support for:
* Embedded SVGs
* SVG images

Pure Java libs:
* Apache Batik
* [SVG Salamander](https://github.com/blackears/svgSalamander)

**Comment by gouessej** [comment]
You can use TwelveMonkeys too, it relies on Batik under the hood.

**Comment by hrj** [comment]
My main concern with Batik is that it has a huge list of dependencies. Atleast the version I looked at did. We don't need the networking capabilities of Batik, which is where most of the dependencies are likely to get pulled in (commons-http et al). I had a crazy idea of patching Batik to only deal with ByteInputStream, and removing any other irrelevant stuff.

**Comment by anii1234** [comment]
i want to contribute in this.

**Comment by hrj** [comment]
@anii1234 Sure, give it a try and send a PR, even if it is just a start. Simple SVG support is sufficient to begin with. I would prefer using SVG Salamander or a modified Batik with fewer dependencies.

**Comment by stellargo** [comment]
JFreeSVG seems to be a good option apparently faster than batik. Is this still up for grab?

**Comment by hrj** [comment]
@stellargo JFreeSVG sounds good. Feel free to pick this up!

**Comment by stellargo** [comment]
Ok I will be working on it !

**Comment by stellargo** [comment]
Well it seems a pretty distorted version of webpages is opening up and google images is not even working

**Comment by hrj** [comment]
Yup, this browser is very much a work-in-progress and layout improvements are on top of the priority list.

But SVG rendering could be implemented pretty independently of the layouting algorithms. You could see how inline images are implemented currently and use a similar approach for inline SVGs.
